uxarray/grid/neighbors.py

```diff
@@ -186,25 +266,35 @@ def query_radius(self,
             raise AssertionError(
                 f"The value of r must be greater than or equal to zero.")
 
-        xyz = _prepare_xyz_for_query(xyz)
+        # Use the correct function to prepare for query based on coordinate type
+        if self.coordinate_type == "cartesian":
+            coords = _prepare_xyz_for_query(coords)
+        elif self.coordinate_type == "spherical":
+            coords = _prepare_xy_for_query(coords, in_radians)
+        else:
+            raise ValueError
 
         if count_only:
-            count = self._current_tree().query_radius(xyz, r, return_distance,
+            count = self._current_tree().query_radius(coords, r,
+                                                      return_distance,
                                                       count_only, sort_results)
 
             return count
 
         else:
 
-            ind, d = self._current_tree().query_radius(xyz, r, return_distance,
+            ind, d = self._current_tree().query_radius(coords, r,
+                                                       return_distance,
                                                        count_only, sort_results)
 
             ind = np.asarray(ind[0], dtype=INT_DTYPE)
 
-            if xyz.shape[0] == 1:
+            if coords.shape[0] == 1:
                 ind = ind.squeeze()
 
             if return_distance:
+                if not in_radians and self.coordinate_type == "spherical":
+                    d = np.rad2deg(d[0])
 
                 return d, ind
```

Comment:
@philipc2 is there a reason we reverse the return variables compared to sklearn here? We return `d, ind` not `ind, d`? Shouldn't we keep with the established return values of the library used?

Reply:
> Shouldn't we keep with the established return values of the library used?

This is a fair point, I don't remember exactly why I chose to swap the order, other than it may have been more intuitive for me.

We can go ahead and change it to be consistent with SKlearn if there's no objections.

Reply:
@philipc2 I see the problem, `query_radius` in SKLearn returns ind, d but `query` returns d, ind. So they aren't consistent across their standards.